docs/examples/Complex Numbers.ipynb

```diff
@@ -0,0 +1,318 @@
+     ]
+    }
+   ],
+   "source": [
+    "for value in np.array([1, -1, 8.2, np.pi]) + 0j:\n",
+    "    try:\n",
+    "        imp(value)\n",
+    "        print(f'Succesfully set the parameter to {value}')\n",
+    "    except TypeError:\n",
+    "        print(f'Sorry, but {value} is not complex')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Array-valued parameters\n",
+    "\n",
+    "There is no separate complex-valued array validator, since the `Arrays` validator can be customized to cover any real or complex valued case.\n",
+    "\n",
+    "Let's make a little array to hold some quantum state amplitudes. Let's pretend to be in a 5-dimensional Hilbert space. Our state parameter should thus hold 5 complex numbers (the state expansion coefficients in some implicit basis)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qcodes.utils.validators import Arrays"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The proper validator should accept complex numbers and should reject anything of the wrong shape. Note that we get to decide whether we want to accept \"non-strictly complex\" data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "amps_val_strict = Arrays(shape=(5,), valid_types=(np.complex128, np.complex64))\n",
+    "amps_val_lax = Arrays(shape=(5,), valid_types=(np.complex128, np.complex64, np.float, np.int))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The strict validator is strict:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Sorry, but integers are not strictly complex\n",
+      "Sorry, but floats are not strictly complex\n"
+     ]
+    }
+   ],
+   "source": [
+    "try:\n",
+    "    amps_val_strict.validate(np.array([1, 2, 3, 4, 5]))\n",
+    "except TypeError:\n",
+    "    print('Sorry, but integers are not strictly complex')\n",
+    "    \n",
+    "try:\n",
+    "    amps_val_strict.validate(np.array([1.0, 2.0, 3.0, 4.0, 5.0]))\n",
+    "except TypeError:\n",
+    "    print('Sorry, but floats are not strictly complex')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "But note that the presence of a single imaginary part will cast the whole array as complex:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Yeah, those are complex numbers\n"
+     ]
+    }
+   ],
+   "source": [
+    "amps_val_strict.validate(np.array([1.0 + 0j, 2.0, 3.0, 4.0, 5.0]))\n",
```

Comment:
should we make it clearer whats going on by doing something along the lines of

```
amp_array = np.array([1.0 + 0j, 2.0, 3.0, 4.0, 5.0])
print(amp_array.dtype)
amps_val_strict.validate(amp_array)
```

Reply:
Yes!